async-download-rest/src/main/java/org/uniprot/api/async/download/messaging/consumer/processor/map/MapRequestProcessor.java

```diff
@@ -0,0 +1,54 @@
+package org.uniprot.api.async.download.messaging.consumer.processor.map;
+
+import static org.uniprot.api.rest.download.model.JobStatus.FINISHED;
+import static org.uniprot.api.rest.download.model.JobStatus.RUNNING;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.springframework.stereotype.Component;
+import org.uniprot.api.async.download.messaging.consumer.processor.RequestProcessor;
+import org.uniprot.api.async.download.messaging.consumer.processor.composite.map.UniProtKBToUniRefMapCompositeRequestProcessor;
+import org.uniprot.api.async.download.model.request.map.MapDownloadRequest;
+import org.uniprot.api.async.download.model.request.map.UniProtKBToUniRefMapDownloadRequest;
+import org.uniprot.api.async.download.service.map.MapJobService;
+
+@Component
+public class MapRequestProcessor implements RequestProcessor<MapDownloadRequest> {
+    protected static final String RESULT_FILE = "resultFile";
+    protected static final String STATUS = "status";
+    public static final String UNIPROT_KB = "UniProtKB";
+    public static final String UNIREF = "UniRef";
+    private final UniProtKBToUniRefMapCompositeRequestProcessor
+            uniProtKBToUniRefMapCompositeRequestProcessor;
+    private final MapJobService jobService;
+
+    public MapRequestProcessor(
+            UniProtKBToUniRefMapCompositeRequestProcessor
+                    uniProtKBToUniRefMapCompositeRequestProcessor,
+            MapJobService jobService) {
+        this.uniProtKBToUniRefMapCompositeRequestProcessor =
+                uniProtKBToUniRefMapCompositeRequestProcessor;
+        this.jobService = jobService;
+    }
+
+    @Override
+    public void process(MapDownloadRequest request) {
+        jobService.update(request.getDownloadJobId(), Map.of(STATUS, RUNNING));
+        processRequest(request);
+        jobService.update(
+                request.getDownloadJobId(),
+                Map.of(STATUS, FINISHED, RESULT_FILE, request.getDownloadJobId()));
+    }
+
+    private void processRequest(MapDownloadRequest request) {
+        if (Objects.equals(UNIPROT_KB, request.getFrom())
+                && Objects.equals(UNIREF, request.getTo())) {
+            uniProtKBToUniRefMapCompositeRequestProcessor.process(
+                    (UniProtKBToUniRefMapDownloadRequest) request);
+        } else {
+            throw new IllegalArgumentException(
```

Comment:
Is it possible to avoid this if else so that we don't have to add another if block another mapping.